addons/repo-proxy/common/src/main/java/org/commonjava/indy/repo/proxy/ContentBrowseRemoteIndyListingRewriteManager.java

```diff
@@ -129,37 +142,63 @@ public boolean rewriteResponse( HttpServletRequest request, HttpServletResponse
 
         logger.trace( "Start to get remote indy content list for indy {} and path {}", remoteIndyUrl, path );
 
-        URL remoteIndyURL = new URL( remoteIndyUrl );
-        String remoteIndyHost = remoteIndyURL.getHost();
-        int remotePort = remoteIndyURL.getPort();
-        // 80 or 443 can be ignored in http(s) url
-        if ( remotePort != 80 && remotePort != 443 && remotePort > 0 )
-        {
-            remoteIndyHost = remoteIndyHost + ":" + remotePort;
-        }
-        String localProxyHost = request.getServerName();
-        int localPort = request.getServerPort();
-        if ( localPort != 80 && localPort != 443 && localPort > 0 )
+        try
         {
-            localProxyHost = localProxyHost + ":" + localPort;
-        }
-        final StoreKey originalStoreKey = StoreKey.fromString( originalRepo.get() );
-        final String remoteContentResult = getRemoteIndyListingContent( originalStoreKey, path );
+            URL remoteIndyURL = new URL( remoteIndyUrl );
+            String remoteIndyHost = remoteIndyURL.getHost();
+            int remotePort = remoteIndyURL.getPort();
+            // 80 or 443 can be ignored in http(s) url
+            if ( remoteIndyUrl.contains( ":" ) && remotePort != 80 && remotePort != 443 && remotePort > 0 )
+            {
+                remoteIndyHost = remoteIndyHost + ":" + remotePort;
+            }
+            String localProxyHost = request.getServerName();
+            int localPort = request.getServerPort();
+            if ( localPort != 80 && localPort != 443 && localPort > 0 )
+            {
+                localProxyHost = localProxyHost + ":" + localPort;
+            }
+            final StoreKey originalStoreKey = StoreKey.fromString( originalRepo.get() );
+            final String remoteContentResult = getRemoteIndyListingContent( originalStoreKey, path );
 
-        if ( StringUtils.isNotBlank( remoteContentResult ) )
+            if ( StringUtils.isNotBlank( remoteContentResult ) )
+            {
+                logger.trace( "Start replacing original host {} to target host {}", remoteIndyHost, localProxyHost );
+                final String replaced =
+                        RepoProxyUtils.replaceAllWithNoRegex( remoteContentResult, remoteIndyHost, localProxyHost );
```

Comment:
:joy: love this method name...that shows some scars

Reply:
Yup, the original String.replaceAll is the evil which caused the regex error in MMENG-593, that's why this method here... :joy: